lib/rules/function-no-unknown/index.js

```diff
@@ -60,39 +64,42 @@ const rule = (primary, secondaryOptions) => {
 
 			if (!isStandardSyntaxValue(value)) return;
 
-			valueParser(value).walk((node) => {
-				const name = node.value;
+			/**
+			 * @param {import('@csstools/css-parser-algorithms').ComponentValue} [componentValue]
+			 */
+			const walker = (componentValue) => {
+				if (!isFunctionNode(componentValue)) return;
 
-				if (node.type !== 'function') {
-					return;
-				}
+				const name = componentValue.getName();
 
-				if (!isStandardSyntaxFunction(node)) {
-					return;
-				}
```

Comment:
Since we've removed this check, is it possible that this rule will now flag errors for SASS/CSS-in-JS users? If so, I imagine that this might be slightly breaking for those users (and/or something we propagate to configs)?

(I could also be missing something about how SASS and CSS-in-JS play together with this rule)

Reply:
Good catch :)

```js
module.exports = function isStandardSyntaxFunction(node) {
	// Function nodes without names are things in parentheses like Sass lists
	if (!node.value) {
		return false;
	}

	if (node.value.startsWith('#{')) {
		return false;
	}

	// CSS-in-JS interpolation
	if (node.value.startsWith('${')) {
		return false;
	}

	// CSS-in-JS syntax
	if (node.value.startsWith('`')) {
		return false;
	}

	return true;
};
```

None of those cases would be parsed as a `FunctionNode` by `@csstools/css-parser-algorithms`.

Only `<ident><open-parenthesis>` (e.g. `foo(`) starts a function in standard CSS.

This behavior makes `isFunctionNode` the equivalent of  `isStandardSyntaxFunction`.